lms/views/api/canvas/sync.py

```diff
@@ -57,11 +57,11 @@ def _to_groups(self, sections):
         context_id = self._request.json["course"]["context_id"]
 
         return [
-            HGroup.section_group(
-                section_name=section["name"],
+            self._grouping_service.canvas_section(
```

Comment:
So whenever the sync API is called we end up upserting all the sections into the DB. The sync API is always used when launching an assignment with sections enabled, so this will always upsert the users sections into the DB whenever launching a sections assignment.

Previously this method used to return a list of `models.HGroup` namedtuples but now it's going to return a list of `models.CanvasSection` groupings, and those will be passed to the sync API. But I guess it still works because `CanvasSection` has the same interface as `HGroup` did.

Some naming tweaks could make it a little clearer what's happening here:

* `self._grouping_service.upsert_canvas_section(...)` would make it clearer that a section is being _upserted_ into the DB
* The method that this line is within is called `_to_groups()` but I think it's actually `_to_groupings()` now?